x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java

```diff
@@ -688,6 +689,12 @@ public static String createDownsampleIndexMapping(
             .utf8ToString();
     }
 
+    private static void addRootSubobjects(Map<String, Object> sourceIndexMappings, XContentBuilder builder) throws IOException {
+        if (sourceIndexMappings.containsKey("subobjects")) {
+            builder.field("subobjects", sourceIndexMappings.get("subobjects"));
+        }
```

Comment:
I'm on the fence for this one, it should be safe but we're updating production logic to make tests happy.. I can add tests if we like it, or guard behind a flag to use in tests only.

Reply:
This looks ok...anyway if you look a bit below we use a `MappingVisitor` object to visit mappings...maybe that is a better way to go through mappings and apply changes. Indeed I wonder if we need to do this for nested `subobjects` properties too.

Reply:
Right, though I was thinking of going to the other direction and skipping objects entirely. Today, we don't propagate any object properties so `subobjects:false` should always be applicable..